script.js

```diff
@@ -1,44 +1,53 @@
-/* eslint-disable no-unused-vars, no-use-before-define */
+/* eslint-disable no-unused-vars, no-use-before-define, class-methods-use-this */
 
 const bookArray = localStorage.getItem('books') ? JSON.parse(localStorage.getItem('books')) : [];
 localStorage.setItem('books', JSON.stringify(bookArray));
 const library = JSON.parse(localStorage.getItem('books'));
 
-function Book(title, author) {
-  this.title = title;
-  this.author = author;
-}
+class Library {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
 
-function createList(title, author) {
-  const li = document.createElement('li');
-  li.textContent = `${title} - by ${author}`;
-  const button = document.createElement('button');
-  button.setAttribute('class', 'rmItem');
-  button.innerHTML = 'Remove';
-  button.addEventListener('click', removeBook);
-  li.appendChild(button);
-  const ul = document.getElementById('bookList');
-  ul.appendChild(li);
-}
+  addBook() {
+    const title = document.getElementById('title').value;
+    const author = document.getElementById('author').value;
+    const newBook = new Library(title, author);
+    bookArray.push(newBook);
+    localStorage.setItem('books', JSON.stringify(bookArray));
+    this.createList(newBook);
+  }
 
-function removeBook(e) {
-  const takeItem = document.querySelectorAll('.rmItem');
-  const bookIndex = bookArray.indexOf.call(takeItem, e.target);
-  bookArray.splice(bookIndex, 1);
-  localStorage.setItem('books', JSON.stringify(bookArray));
-  const updatedLibrary = JSON.parse(localStorage.getItem('books'));
-  document.getElementById('bookList').innerHTML = '';
-  updatedLibrary.forEach((item) => { createList(item.title, item.author); });
-}
+  removeBook(e) {
+    const takeItem = document.querySelectorAll('.rmItem');
+    const bookIndex = bookArray.indexOf.call(takeItem, e.target);
+    bookArray.splice(bookIndex, 1);
```

Comment:
**OPTIONAL**

It will be nice if you can favor the use of array methods that doesn't mutate the array, e.g. `concat`, `filter`, `slice` instead of `push`, `pop`, `splice` and so on. [Learn more...](https://lorenstewart.me/2017/01/22/javascript-array-methods-mutating-vs-non-mutating/)